src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'Blue',
+}
+
+enum Shape {
+  Circle = 'circle',
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  color: Color,
+  shape: Shape,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Circle implements Figure {
+  public shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+  ) {
+    if (a <= 0) {
+      throw new Error('value can`t be 0 and <');
+    }
+  }
 
+  getArea(): number {
+    return Math.trunc((Math.PI * (this.a ** 2)) * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('value can`t be 0 and <');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('a + b can`t be c');
+    }
```

Comment:
Why? it's not very obvious what's wrong. it would be better
`sides 1, 2 and 3 can not form a triangle`
or
`sum of cathets can`t be less then hypotenuse`